### PR TITLE
drivers: sensor: add driver for Invensense ICM42670 motion sensor

### PR DESCRIPTION
**Pull request by aunsbjerg**
Adds support for the Invensense ICM42670 6-axis motion sensor with support for accelerometer, gyroscope and on-die temperature readings.

Driver work is based on the ICM42605 (#29682) which is quite similar but still different enough in register layout and register access methods that it warrants a separate driver. The ICM42670 driver does not include support for the 2K sample FIFO on the IC, instead only "normal" data ready interrupts are supported.

Currently implemented features are:

* trigger support for data ready interrupts
* accelerometer and gyroscope sample frequency and range attribute setting

Future work includes alarm trigger support, alarm threshold attribute setting, power management and low power mode support, and support for advanced features like pedometer, freefall detection, tilt detection etc.

### CODE REVIEW

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Uses of `default` in devicetree bindings must be justified in the descriptions for each property with a default value.

Please read https://docs.zephyrproject.org/latest/guides/dts/bindings.html#default-values-for-properties and fix the bindings accordingly.

**Comment by aunsbjerg** [comment]
> Uses of `default` in devicetree bindings must be justified in the descriptions for each property with a default value.
> 
> Please read https://docs.zephyrproject.org/latest/guides/dts/bindings.html#default-values-for-properties and fix the bindings accordingly.

Removed default values and converted to required properties instead.

**Review by MaureenHelm — DISMISSED** [review]
Looks great, thanks

File: drivers/sensor/icm42670/Kconfig [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# ICM42670 Six-Axis Motion Tracking device configuration options
+#
+# Copyright (c) 2022 Esco Medical ApS
+# Copyright (c) 2020 TDK Invensense
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig ICM42670
+	bool "ICM42670 Six-Axis Motion Tracking Device"
+	depends on SPI
```
[the remark this answers is not in the document]
Comment (aunsbjerg) [reply drivers/sensor/icm42670/Kconfig:13]: Neat, hadn't seen that before. Added as suggested, but keeping the explicit `CONFIG_ICM42670=y` in the build_all prj.conf to keep with the existing pattern.

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**